app/controllers/concerns/foreman/controller/provisioning_templates.rb

```diff
@@ -19,7 +19,7 @@ def default_template_url(template, hostgroup)
     protocol = uri.scheme
 
     url_for(:only_path => false, :action => :hostgroup_template, :controller => '/unattended',
-            :id => template.name, :hostgroup => hostgroup.name, :protocol => protocol,
+            :id => template.name, :hostgroup => hostgroup.title, :protocol => protocol,
```

Comment:
The title attribute includes the parents's host group,  so hostgroups like `production/example` and `test/example`  will generate distinct url.